pkg/cached/cache.go

```diff
@@ -183,10 +183,27 @@ func NewListMerger[T, V any](mergeFn func(results []Result[T]) Result[V], caches
 		caches:  caches,
 	}
 }
+
 func (c *listMerger[T, V]) prepareResults() []Result[T] {
-	cacheResults := make([]Result[T], 0, len(c.caches))
-	for _, cache := range c.caches {
-		cacheResults = append(cacheResults, cache.Get())
+	cacheResults := make([]Result[T], len(c.caches))
+	ch := make(chan struct {
+		int
+		Result[T]
+	}, len(c.caches))
+	for i := range c.caches {
+		go func(index int) {
+			ch <- struct {
+				int
+				Result[T]
+			}{
+				index,
+				c.caches[index].Get(),
+			}
+		}(i)
+	}
+	for i := 0; i < len(c.caches); i++ {
+		res := <-ch
+		cacheResults[res.int] = res.Result
```

Comment:
Out of curiosity, is this single thread safe? Is it a good time to revisit our atomicity guarantees for the library?

Reply:
I don't think this changes the guarantees that are already documented. Do you suggest that we need more tests to verify these guarantees?

Reply:
Not being thread safe usually means that we still have a guarantee for correctness when the function is not called in parallel. Spawning threads to call a Get() that is not thread safe seems like it may break that guarantee?